hts.c

```diff
@@ -866,6 +871,11 @@ htsFile *hts_open_format(const char *fn, const char *mode, const htsFormat *fmt)
         *mode_c = format_to_mode[fmt->format];
     }
 
+    // Uncompressed bam/bcf is not supported, remove 'u' flag on write
+    if (uncomp && *mode_c == 'b' && (strchr(smode, 'w') || strchr(smode, 'a'))) {
+        memcpy(uncomp, uncomp + 1, cp2 - uncomp - 1);
```

Comment:
You should use `memmove()` here because the source and destination buffers overlap, and [`memcpy()` disallows that](https://pubs.opengroup.org/onlinepubs/9699919799/functions/memcpy.html).  But better still would be to turn the 'u' into a '0', as the result will then be uncompressed while still BGZF wrapped.

Reply:
Updated accordingly.